File: src/modules/user/services/user_find_self_service.py
from src.modules.user.models import User
from src.shared.types.user_request import UserRequest
from src.shared.exceptions.not_found_exception import NotFoundException


class UserFindSelfService:
    @classmethod
    def execute(self, user_request: UserRequest):

        user = User.objects.filter(id=user_request.id, deleted_at=None).first()

        if user is None:
            raise NotFoundException("Usuário não encontrado!")

        return user

File: src/modules/user/services/user-find-self.service.ts
import { prisma } from '../../../infra/database/prisma.service';
import { NotFoundException } from '../../../shared/exceptions';
import { UserRequest } from '../../../shared/types/user-request';
import { UserEntity } from '../domain/entities/user.entity';

class UserFindSelfService {
  async execute(userRequest: UserRequest): Promise<UserEntity> {
    if (!userRequest) throw new NotFoundException('Usuário não encontrado!');

    const user = await prisma.user.findFirst({
      where: {
        id: userRequest.id,
        deletedAt: null,
      },
      include: {
        role: true,
        organization: true,
        company: true,
        avatar: true,
      },
    });

    if (!user) throw new NotFoundException('Usuário não encontrado!');

    return new UserEntity(user);
  }
}

export const userFindSelfService = new UserFindSelfService();
